code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,183 @@
+		new /datum/GBP_equipment("Ranged RCD x3",			/obj/item/weapon/rcd/arcd,										6000,	3),
+		new /datum/GBP_equipment("ERT Hardsuit x5",		/obj/item/clothing/suit/space/hardsuit/ert/engi,					7000,	5),
+		new /datum/GBP_equipment("Portal Gun x5",			/obj/item/weapon/gun/energy/wormhole_projector,					8000,	5),
+		new /datum/GBP_equipment("Prototype Repair Vehicle x2",		/obj/vehicle/space/speedbike/repair,					10000,	2),
+		new /datum/GBP_equipment("Reactive Decoy Armor x5",		/obj/item/clothing/suit/armor/reactive/stealth,				12500,	5),
+		new /datum/GBP_equipment("Chrono Suit x5",		/obj/item/clothing/suit/space/chronos,								20000,	5),
+		new /datum/GBP_equipment("WHAT HAVE YOU DONE... x5",		/obj/vehicle/space/speedbike/memewagon,				30000,	5),
+		)
+
+/datum/GBP_equipment
+	var/equipment_name = "generic"
+	var/equipment_path = null
+	var/cost = 0
+	var/amount = 0
+
+/datum/GBP_equipment/New(name, path, cost, amount)
+	src.equipment_name = name
+	src.equipment_path = path
+	src.cost = cost
+	src.amount = amount
+
+/obj/machinery/engi_points_manager/Initialize()
+	radio = new(src)
+	radio.listening = 0
+	radio.frequency = 1357
+	..()
+
+/obj/machinery/engi_points_manager/Destroy()
+	if(radio)
+		qdel(radio)
+		radio = null
+	return ..()
+
+
+/obj/machinery/engi_points_manager/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/engi_points_manager/update_icon()
+	if(powered())
+		icon_state = initial(icon_state)
+	else
+		icon_state = "[initial(icon_state)]-off"
+
+/obj/machinery/engi_points_manager/attack_hand(mob/user)
+	if(..())
+		return
+	interact(user)
+
+/obj/machinery/engi_points_manager/interact(mob/user)
+	var/dat
+	dat +="<div class='statusDisplay'>"
+	dat += "You currently have <td>[round(GBP)]</td> engineering voucher points<br>"
+	dat += "You have earned a total of <td>[round(GBPearned)]</td> this shift<br>"
+	dat += "</div>"
+	dat += "<br><b>Equipment point cost list:</b><BR><table border='0' width='300'>"
+	for(var/datum/GBP_equipment/prize in prize_list)
+		dat += "<tr><td>[prize.equipment_name]</td><td>[prize.cost]</td><td><A href='?src=\ref[src];purchase=\ref[prize]'>Purchase</A></td></tr>"
+	dat += "</table>"
+
+	var/datum/browser/popup = new(user, "vending", "Engineering Point Redemption", 400, 350)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/engi_points_manager/Topic(href, href_list)
+	if(..())
+		return
+	if(href_list["purchase"])
+		var/datum/GBP_equipment/prize = locate(href_list["purchase"])
+		if (!prize || !(prize in prize_list))
+			return
+		if(prize.cost > GBP)
+			return
+		else if(prize.cost <= GBP) // Placeholder spaghetti calm your shit
+			GBP -= prize.cost
+			for(var/i in 1 to prize.amount)
+				for(var/obj/machinery/engi_points_delivery/D in machines)
+					D.icon_state = "geardist-load"
+					playsound(D, 'sound/machines/Ding.ogg', 100, 1)
+					spawn(20)
+						new prize.equipment_path(get_turf(D))

```

Comment:
And

>https://github.com/tgstation/tgstation/blob/d5c870e7148f8cc518c5eae6cf4cdf191bd4b361/code/__HELPERS/mobs.dm#L343
please use this spawn helper so you dont' accidentally spawn thigns into nullspace and we get a runtiem if it does occur